base/linking.jl

```diff
@@ -102,7 +102,7 @@ function ld()
     elseif Sys.isapple()
         flavor = "darwin"
         arch = Sys.ARCH == :aarch64 ? :arm64 : Sys.ARCH
-        default_args = `-arch $arch -U dyld_stub_binding_helper -U dyld_stub_binder -platform_version macos $(Base.MACOS_PRODUCT_VERSION) $(Base.MACOS_PLATFORM_VERSION)`
+        default_args = `-arch $arch -undefined dynamic_lookup -platform_version macos $(Base.MACOS_PRODUCT_VERSION) $(Base.MACOS_PLATFORM_VERSION)`
```

Comment:
Alternative might be `-flat_namespace` which should use the `-lSystem` dependency of `-ljulia` and lookups symbols there. But I don't know if that would work on systems without xcode.

I have a mac that I can verify this on tomorrow.

Reply:
Did you ever verify this?

Reply:
What exactly do you want to verify? Perhaps I can help, I have a macOS VM w/o Xcode around.

I am not sure what exactly you think might not work with Xcode missing?

Reply:
If replacing `-undefined dynamic_lookup` with `-flat_namespace` work when libSystem is not available due to Xcode not being installed.